### PR TITLE
Caller docs and fixes

### LINKED ISSUE #526: caller.jl should be documented and fixed
There is currently no way to to call a Singular library procedure that take 0 arguments in the Singular langugage *and* takes the hidden argument `basering`. This problem should be fixed, and it should be documented how Singular.jl determines the `basering` in such library calls.

### PR DESCRIPTION
**Pull request by tthsqe12**
fixes #526

### CODE REVIEW

File: docs/src/caller.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+ - The singular language allows arbitrary attributes to be attached to each
+   object. These are not supported because Singular.jl works with the raw
+   kernel types ring, poly, ideal, etc. that do not have attributes.
+
+ - Maps between polynomial rings cannot be effectively communicated between the
+   Singular language and Singular.jl because the Singular language tracks the
+   preimage ring by name only and the rings in Singular.jl do not have names.
+
+ - Very few library proceedures work with [Nemo rings and fields](@ref)
```
Comment (thofma) [inline docs/src/caller.md:19]:
```suggestion
 - Very few library proceedures work with [Nemo rings and fields](@ref).
```
If I have nothing to say, I can always point out punctuation marks.

File: docs/src/caller.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+corresponding function in `Singular.jl` is called `Singular.LibGeneral.sort`.
+The full list of libraries included can be viewed by typing `Singular.Lib` at
+the REPL and double pressing the tab key.
+
+One issue that comes up in calling library proceedures is the implicit argument
+`basering` that all procedures receive in the Singular language. `Singular.jl`
+tries to infer the base ring from the arguments provided to the function. When
+this fails or is simply not possible, the user can always provide a base ring
+by passing it in as the first argument to the `Singular.jl` function. Note that
+if the first argument to the `Singular.jl` version of a library procedure, then
```
Comment (thofma) [inline docs/src/caller.md:33]:
```suggestion
if the first argument to the `Singular.jl` version of a library procedure, then
```
If the first argument is what?

File: docs/src/noncommutative.md [hunk 3]
```diff
@@ -52,26 +52,36 @@ $D$ over $R$. If the variables of $R$ are $x_1,\dots,x_n$, then the noncommutati
 algebra is constructed with relations $x_j x_i = c_{i,j} x_i x_j + d_{i,j}$ for
 $1 \le i < j \le n$. The $c_{i,j}$ must be nonzero constant polynomials and the
 relations $x_i x_j > \mathrm{lm}(d_{i,j})$ must hold in the monomial ordering
-of the ring $R$. The entries of the matrices $C$ and $D$ on or
-below the main diagonal are ignored.
+of the ring $R$.
+
+The entries of the matrices $C$ and $D$ on or below the main diagonal are
+ignored. A non-matrix argument `a` for either `C` or `D` is turned into a
+matrix with with all relevent entries set to `a`.
```
Comment (fingolfin) [inline docs/src/noncommutative.md:59]:
```suggestion
matrix with all relevent entries set to `a`.
```

**Review by fingolfin — APPROVED** [review]
minor nitpick

**Pull request merged**